workspaces/shape-hash/src/json-to-shape-hash.ts

```diff
@@ -76,10 +76,12 @@ function toJson(item: any) {
   switch (item.type) {
     case types.OBJECT:
       const newObj = {};
-      item.fields.forEach(({ key, hash }: any) => {
-        //@ts-ignore
-        newObj[key] = toJson(hash);
-      });
+      if (item.hasOwnProperty('fields')) { 
```

Comment:
Is there a better way to do this? Perhaps a variant of the `forEach`?

Reply:
you can use item.fields.reduce(accumulator, newObj), but if it's possible for types.OBJECT to have .fields be undefined then you need the hasOwnProperty check. I believe the //@ts-ignore could be removed if newObj were typed appropriately as well

Reply:
👍  done. I'll merge once the tests pass.